bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,913 @@
+                        request.header("Accept", "application/json");
+                        if (csrf != null) {
+                            request.header("X-CSRF-TOKEN", csrf);
+                        }
+                        if (!data.equals("empty")) {
+                            request.content(new BytesContentProvider(data.getBytes("UTF-8")),
+                                    "application/x-www-form-urlencoded; charset=UTF-8");
+                        } else {
+                            logger.debug("Setting cookie with username {} and vid {}", userName, password);
+                            request.cookie(new HttpCookie("username", userName));
+                            request.cookie(new HttpCookie("vid", password));
+                        }
+                        logger.debug("HTTP POST Request {}.", request.toString());
+                        response = request.send();
+                        content = response.getContentAsString();
+                        contentUTF8 = new String(content.getBytes("UTF-8"), "ISO-8859-1");
+                        if (response.getStatus() == HttpStatus.OK_200) {
+                            if (content.contains("\"message\":\"Request Failed. Code 503 from")) {
+                                logger.warn("Failed to send HTTP POST due to {}", contentUTF8);
+                                return HttpStatus.SERVICE_UNAVAILABLE_503;
+                            } else {
+                                setApiServerInUse(otherApiServer);
+                            }
+                        }
+                    }
+                }
+                logger.debug("HTTP Response ({})", response.getStatus());
+                logger.trace("Response body: {}", contentUTF8);
+                return response.getStatus();
+            } catch (ExecutionException e) {
+                logger.warn("Caught ExecutionException {}", e);
+            } catch (UnsupportedEncodingException e) {
+                logger.warn("Caught UnsupportedEncodingException {}", e);
+            } catch (InterruptedException e) {
+                logger.warn("Caught InterruptedException {}", e);
+            } catch (TimeoutException e) {
+                logger.warn("Caught TimeoutException {}", e);
+            }
+        }
+        return 999;
+    }
+
+    private int setSessionCookieAuthLogin() {
+        // URL to set status which will give us 2 cookies with username and password used for the session
+        String url = STATUS;
+
+        try {
+            ContentResponse response = httpClient.GET(url);
+            logger.trace("HTTP Response ({}) Body:{}", response.getStatus(),
+                    response.getContentAsString().replaceAll("\n+", "\n"));
+            CookieStore c = httpClient.getCookieStore();
+            List<HttpCookie> cookies = c.get(URI.create("http://verisure.com"));
+            Iterator<HttpCookie> cookiesIterator = cookies.iterator();
```

Comment:
You could simplify this by forEach lambda

cookies.forEach(cookie -> {

Reply:
I'll fix!